07-http_gordle/6_6_query_param/internal/handlers/guess/handler.go

```diff
@@ -9,67 +9,66 @@ import (
+	"learngo-pockets/httpgordle/internal/api"
 	"learngo-pockets/httpgordle/internal/gordle"
-	"learngo-pockets/httpgordle/internal/handlers/apiconversion"
 	"learngo-pockets/httpgordle/internal/repository"
 	"learngo-pockets/httpgordle/internal/session"
 )
 
 type gameGuesser interface {
 	Find(session.GameID) (session.Game, error)
-	Update(session.GameID, session.Game) error
+	Update(game session.Game) error
 }
 
-// Handler returns a handler for guess requests.
-func Handler(repo gameGuesser) http.HandlerFunc {
-	return func(writer http.ResponseWriter, request *http.Request) {
+// Handle returns the handler for the game creation endpoint.
+func Handle(db gameGuesser) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
 		// Read the Game ID from the query parameters.
-		id := chi.URLParam(request, api.GameID)
+		id := chi.URLParam(req, api.GameID)
 		if id == "" {
-			http.Error(writer, "missing the id of the game", http.StatusBadRequest)
+			http.Error(w, "missing the id of the game", http.StatusNotFound)
 			return
 		}
 
 		// Read the request, containing the guess, from the body of the input.
 		r := api.GuessRequest{}
-		err := json.NewDecoder(request.Body).Decode(&r)
+		err := json.NewDecoder(req.Body).Decode(&r)
 		if err != nil {
-			http.Error(writer, err.Error(), http.StatusBadRequest)
+			http.Error(w, err.Error(), http.StatusBadRequest)
 			return
 		}
 
-		game, err := play(repo, session.GameID(id), r.Guess)
+		game, err := guess(session.GameID(id), r.Guess, db)
 		if err != nil {
 			switch {
 			case errors.Is(err, repository.ErrNotFound):
-				http.Error(writer, err.Error(), http.StatusNotFound)
-			case errors.Is(err, gordle.ErrInvalidGuessLength):
-				http.Error(writer, err.Error(), http.StatusBadRequest)
+				http.Error(w, err.Error(), http.StatusNotFound)
+			case errors.Is(err, gordle.ErrInvalidGuess):
+				http.Error(w, err.Error(), http.StatusBadRequest)
 			case errors.Is(err, session.ErrGameOver):
-				http.Error(writer, err.Error(), http.StatusForbidden)
+				http.Error(w, err.Error(), http.StatusForbidden)
 			default:
-				http.Error(writer, err.Error(), http.StatusInternalServerError)
+				http.Error(w, err.Error(), http.StatusInternalServerError)
 			}
 			return
 		}
 
-		apiGame := apiconversion.ToAPIResponse(game)
+		apiGame := api.ToGameResponse(game)
 
-		writer.Header().Set("Content-Type", "application/json")
-		err = json.NewEncoder(writer).Encode(apiGame)
+		w.Header().Set("Content-Type", "application/json")
+		err = json.NewEncoder(w).Encode(apiGame)
 		if err != nil {
-			// The header has already been set. Nothing much we can do here.
-			log.Printf("failed to write response: %s", err)
+			http.Error(w, "failed to write response", http.StatusInternalServerError)
+			return
 		}
 	}
 }
 
-func play(repo gameGuesser, id session.GameID, guess string) (session.Game, error) {
+func guess(id session.GameID, guess string, db gameGuesser) (session.Game, error) {
```

Comment:
why would `gameGuesser` be a `db`?  I'd rename the variable to `gg` or `guesser`

Reply:
let's use guesser